src/categorical_algebra/CSets.jl

```diff
@@ -1619,20 +1619,21 @@ subparts.
 """
 function abstract_attributes(X::ACSet)
   S = acset_schema(X)
-  A = copy(X); 
-  comps = Dict{Any,Any}(map(attrtypes(S)) do at
-    rem_parts!(A, at, parts(A,at))
-    comp = Union{AttrVar,attrtype_type(X,at)}[]
+  A = deepcopy(X);
```

Comment:
Thanks for the fix! I doubt this needs to be a deep copy since you're not mutating the attribute *values*, just the acset parts which are copied by a shallow copy.

Reply:
ACSets.jl actually already does a `deepcopy` on the parts (and a shallow copy of the subparts), so switching this to `copy` does have the correct behavior.

Reply:
Yay!